BridgeAppSDK/SBAGenericPageStepViewController.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  SBAGenericPageStepViewController.swift
+//  ResearchUXFactory
+//
+//  Created by Josh Bruhin on 6/13/17.
+//  Copyright © 2017 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+
+import UIKit
+
+class SBAGenericPageStepViewController: ORKPageStepViewController, UIPageViewControllerDelegate {
```

Comment:
Yeah, it's a bit more complicated than that. You need to check that the view controller to be returned *should* be a generic *and* you need to check that the steps are a straight line and do not include navigation that would make your method of determining step count inaccurate. That's why the ResearchKit version of the task view controller progress asks the task rather than just counting steps.

For now, just explicitly use it for consent. :)

*OR* you can have `SBAGenericStepViewController` call the method on `self.step.task?.progress` (or something like that, can't remember what it's named and don't have time at the moment to code hunt.  That should work b/c internally the `ORKPageStep` is backed by a subclass of ORKOrderedTask.